src/xdsl/ir.py

```diff
@@ -778,6 +795,25 @@ def erase_block(self, block: Union[int, Block], safe_erase=True) -> None:
         block = self.detach_block(block)
         block.erase(safe_erase=safe_erase)
 
+    def clone_into(self, dest: Region, mapper: dict):
```

Comment:
Same thing here with the mapper. Also, 'dict' is python 3.10 specific, could we use `Dict` here instead?